agent/agent.go

```diff
@@ -618,29 +623,37 @@ func (a *Agent) endTaskResponse(tc *taskContext, status string, message string)
 	return detail
 }
 
-func (a *Agent) runPostTaskCommands(ctx context.Context, tc *taskContext) {
+func (a *Agent) runPostTaskCommands(ctx context.Context, tc *taskContext) error {
 	start := time.Now()
 	a.killProcs(ctx, tc, false)
 	defer a.killProcs(ctx, tc, false)
 	tc.logger.Task().Info("Running post-task commands.")
+	opts := runCommandsOptions{}
 	postCtx, cancel := a.withCallbackTimeout(ctx, tc)
 	defer cancel()
 	taskConfig := tc.getTaskConfig()
 	taskGroup, err := taskConfig.GetTaskGroup(tc.taskGroup)
 	if err != nil {
 		tc.logger.Execution().Error(errors.Wrap(err, "error fetching task group for post-task commands"))
-		return
+		return nil
 	}
 	if taskGroup.TeardownTask != nil {
-		err = a.runCommands(postCtx, tc, taskGroup.TeardownTask.List(), runCommandsOptions{})
-		tc.logger.Task().Error(message.WrapError(err, message.Fields{
-			"message": "Error running post-task command.",
-		}))
+		opts.shouldSetupFail = taskGroup.TeardownGroupFailTask
+		err = a.runCommands(postCtx, tc, taskGroup.TeardownTask.List(), opts)
+		if err != nil {
+			tc.logger.Task().Error(message.WrapError(err, message.Fields{
```

Comment:
Can you specify where the double logging occurs? It looks like runCommands only returns an error and doesn't log it out, and this log message has been in the codebase for some time.

Reply:
Oh I think I said the wrong thing but there is still double logging, because we're logging here, and we're logging in FinishTask if runPostTaskCommands returns an error (I think we can just remove that log)